tests/acceptance/pageObjects/FilesPageElement/sharingDialog.js

```diff
@@ -147,6 +147,29 @@ module.exports = {
         .click(informationSelector)
         .waitForElementVisible(selectRoleButton)
     },
+    /**
+     *
+     * @param {string} permission
+     * @param {bool} enable
+     */
+    changeSinglePermission: async function (permission, enable) {
+      const permissionXpath = this.getPermissionSwitchXpath(permission)
```

Comment:
As you are already refactoring, can you change `permissonButton.selector` xpath to `//ul[@class="uk-accordion"]//li[contains(@class, "uk-open")]//span[.="Can %s"]/parent::div/div`.

This will only return permission button whose collbaorator's list is visible.